src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -543,7 +550,17 @@ private async Task CheckJobCompletionStatus()
                 // all queries marked as complete, reindex job is done, check success or failure
                 if (_reindexJobRecord.QueryList.Keys.All(q => q.Status == OperationStatus.Completed))
                 {
-                    await UpdateParametersAndCompleteJob();
+                    // Perform a final check to make sure there are no resources left to reindex
+                    int totalCount = await CalculateTotalCount();
+                    if (totalCount != 0)
+                    {
+                        string message = $"{totalCount} resources failed to be reindexed.";
```

Comment:
Do we want to explain to the user that they should resubmit the same reindex job again in order to finish indexing the remaining resources?

Reply:
Good idea - updated the error message returned to the user in f1bce27.

![2021-07-19 15_03_59-Response(47ms) - Visual Studio Code](https://user-images.githubusercontent.com/54082711/126233588-60a4ac96-2f0b-4fdf-a72d-49b2fd6f8af8.png)